src/main/java/toughasnails/block/BlockRainCollector.java

```diff
@@ -139,14 +139,9 @@ else if (item == TANItems.canteen)
             {
             	if (i > 0 && !worldIn.isRemote)
                 {
-                    if (!playerIn.capabilities.isCreativeMode)
-                    {
-                        heldItem.setItemDamage(3);
-                    }
-
-                    this.setWaterLevel(worldIn, pos, state, i - 1);
-                }
-            	
+            		return ((ItemCanteen) item).onItemRightClick(worldIn, playerIn, hand)
+            			.getType().equals(EnumActionResult.SUCCESS) ? true : false; 
+                }            	
```

Comment:
We use spaces, not tabs. Also the line wrapping isn't too great, better off keeping it as an editor setting, not putting it into the actual code